Library/Homebrew/formula.rb

```diff
@@ -1625,6 +1625,50 @@ def extract_macho_slice_from(file, arch = Hardware::CPU.arch)
   end
   private :extract_macho_slice_from
 
+  # Generate shell completions for a formula for bash, zsh, and fish, using the formula's executable.
+  #
+  # @param base_name [String] the base name of the generated completion script. Defaults to the formula name.
+  # @param shells [Array<Symbol>] the shells to generate completion scripts for. Defaults to `[:bash, :zsh, :fish]`.
+  # @param executable [Pathname] the executable to use for generating the completion scripts. Defaults to the
```

Comment:
Kinda feel like this should be mandatory, we tend to not call executables implicitly without naming them as it makes the DSL less readable.